core/azure-core/src/main/java/com/azure/core/polling/Poller.java

```diff
@@ -0,0 +1,240 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.polling;
+
+import reactor.core.Disposable;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+/**
+ * This is implementation for Long Running Operations.
+ */
+public class Poller<T> {
+
+    /*
+     * pollOperation is a Function that takes the previous PollResponse, and
+     * returns a new PollResponse to represent the current state
+     */
+    private Function<PollResponse<T>, PollResponse<T>> pollOperation;
+
+    /*
+     * Various configuration options to create poller object.
+     */
+    private PollerOptions pollerOptions;
+
+    /*
+     * This will save last poll response.
+     */
+    private PollResponse<T> pollResponse;
+
+    /*
+     * This will be called when cancel operation is triggered.
+     */
+    private Consumer<Poller> cancelOperation;
+
+    /*
+     * Indicate to poll automatically or not when poller is created.
+     * default value is false;
+     */
+    private boolean autoPollingEnabled;
+
+    /*
+     * Indicate if cancel is initiated.
+     */
+    private boolean cancelInitiated;
+
+    private Flux<PollResponse<T>> fluxHandle;
+
+    /*
+     * Since constructor create a subscriber and start autopoll.
+     * This subscriber will be duplicate when client call subscriber on Flux.
+     * Thus this handle will be used to dispose the subscriber when
+     * client invoke poll function
+     */
+    private Disposable fluxDisposable;
+
+    /**
+     * Create a Poller object. Auto polling is turned on by default.
+     * @param pollerOptions configuration options for poller.
+     * @param pollOperation This is the operation to be called.
+     */
+    public Poller(PollerOptions pollerOptions, Function<PollResponse<T>, PollResponse<T>> pollOperation) {
```

Comment:
The poll operation should probably be async? Like `Function<PollResponse<T>, Mono<PollResponse<T>>>`?